sktime/tests/test_switch.py

```diff
@@ -0,0 +1,43 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Switch utility for determining whether tests for a class should be run or not."""
+
+__author__ = ["fkiraly"]
+
+
+def run_test_for_class(cls):
+    """Check if test should run for a class.
+
+    This checks the following conditions:
+
+    * if required dependencies are not present, does not run the test
+    * if ONLY_CHANGED_MODULES setting is on, runs the test
+      if and only if the class has changed according ot is_class_changed
```

Comment:
```suggestion
      if and only if the module containing the class has changed according to is_class_changed
```

Minor text change suggestion. Unless this is intentional with future plan of improving `is_class_changed` to actually detect change in a single class?

Reply:
no, it is meant to refer to the module, indeed.

In the future, I was thinking to optionally retrieve parent classes, i.e., test all classifiers if we change the `BaseClassifier`.